include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp

```diff
@@ -681,7 +645,7 @@ void pclomp::MultiGridNormalDistributionsTransform<PointSource, PointTarget>::co
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template<typename PointSource, typename PointTarget>
-void pclomp::MultiGridNormalDistributionsTransform<PointSource, PointTarget>::updateHessian(Eigen::Matrix<double, 6, 6> &hessian, const Eigen::Matrix<double, 3, 6> &point_gradient, const Eigen::Matrix<double, 18, 6> &point_hessian, const Eigen::Vector3d &x_trans, const Eigen::Matrix3d &c_inv) const {
+void pclomp::MultiGridNormalDistributionsTransform<PointSource, PointTarget>::updateHessian(Eigen::Matrix<double, 6, 6> &hessian, const Eigen::Matrix<double, 4, 6> &point_gradient, const Eigen::Matrix<double, 24, 6> &point_hessian, const Eigen::Vector3d &x_trans, const Eigen::Matrix3d &c_inv) const {
```

Comment:
Please let me confirm my understanding.
updateHessian is only used from computeHessian, and computeHessian is only used in the following lines.
https://github.com/tier4/ndt_omp/blob/01ae2426b72ac791bd3695aa9d0660410f1af426/include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp#L937

This is a branch when something is updated in line_search, and since line_search is currently turned off, step_iteration is always 0, so I don't think computeHessian will be called.

Is my understanding correct?
In other words, even if there is a bug in updateHessian, it will not be noticed in its current behavior, and I am concerned that it will occur someday when line_search is turned on.

Reply:
@SakodaShintaro If the line search is turned on, errors will 100% occur. It is because the angular derivatives (j_ang_a_, j_ang_b, etc) are not pre-computed. Those variables are used by the computeHessian.

Reply:
@SakodaShintaro If the line search is turned on, errors will 100% occur. It is because the angular derivatives (j_ang_a_, j_ang_b, etc) are not pre-computed. Those variables are used by the computeHessian.

Reply:
@anhnv3991 
I would like to maintain line_search to be able to use.

Steps

(1) Organize parameters: this PR https://github.com/tier4/ndt_omp/pull/43
(2) Add a bool type parameter called use_line_search and make it possible to switch true/false: next PR
(3) I would like to merge this PR while maintaining a buildable state.